recipes-bsp/firmware-imx/firmware-imx_7.6.bb

```diff
@@ -76,6 +76,27 @@ do_install() {
     find ${D}${base_libdir}/firmware/ -name '*.mk' -exec rm '{}' ';'
 }
 
+do_deploy() {
+}
+do_deploy_append_mx8mq() {
+    # Synopsys DDR
+    for ddr_firmware in ${DDR_FIRMWARE_NAME}; do
+        install -m 0644 ${S}/firmware/ddr/synopsys/${ddr_firmware} ${DEPLOYDIR}
+    done
+    install -m 0644 ${S}/firmware/hdmi/cadence/signed_hdmi_imx8m.bin ${DEPLOYDIR}
+}
+do_deploy_append_mx8qm() {
+    # Cadence HDMI
+    install -m 0644 ${S}/firmware/hdmi/cadence/hdmitxfw.bin ${DEPLOYDIR}
+    install -m 0644 ${S}/firmware/hdmi/cadence/hdmirxfw.bin ${DEPLOYDIR}
+    install -m 0644 ${S}/firmware/hdmi/cadence/dpfw.bin ${DEPLOYDIR}
+}
+do_deploy_append_mx8qxp() {
+    # SECO
+    install -m 0644 ${S}/firmware/seco/ahab-container.img ${DEPLOYDIR}
+}
+addtask deploy before do_build after do_install
```

Comment:
Please split those in specific recipes so we can build / install just the need and not clobber the deploy directory with unrelated files.

Reply:
The deploy directory is machine-specific, so not sure what you mean here.

$ find tmp/deploy/ -name ahab-container.img
tmp/deploy/images/imx8qxpmek/ahab-container.img